rainier-compute/src/main/scala/com/stripe/rainier/compute/Cholesky.scala

```diff
@@ -0,0 +1,86 @@
+package com.stripe.rainier.compute
+
+/*
+This represents an NxN matrix A
+by the  N(N+1)/2 packed lower-triangular elements
+of its Cholesky decomposition L
+(where A = LL*)
+*/
+case class Cholesky(packed: Vector[Real]) {
+    val rank: Int = Cholesky.rank(packed.size)
+    
+    //log(det(A))
+    def logDeterminant: Real = Real.sum(diagonals.map(_.log)) * 2
+
+    //solve for x where Ax=y
+    def inverseMultiply(y: Vector[Real]): Vector[Real] = {
+        //L*Lx = y, let b=Lx 
+        //solve L*b = y
+        val b = Cholesky.upperTriangularSolve(packed, y)
+        //solve Lx = b
+        Cholesky.lowerTriangularSolve(packed, b)
+    }
+
+    //diagonal elements of L
+    private def diagonals: Vector[Real] = 
+        0.until(rank).toVector.map{i =>
+            packed(Cholesky.triangleNumber(i))
+        }
+}
+
+object Cholesky {
+    def lowerTriangularMultiply(packed: Array[Double], vector: Array[Double]): Array[Double] = {
+        val n = vector.size
+        val result = new Array[Double](n)
+        var i = 0
```

Comment:
seems like doing a `require` on the sizes would be helpful and not impact performance (something like:

```scala
require(packed.size == (vector.size * (vector.size + 1) / 2), s"expected sizes to be consistent: packed = ${packed.size} vector = ${vector.size})
```